File: 1 lab/j.cpp
#include <iostream>
#include <deque>

using namespace std;

int main(){
    char method;
    deque<int> deq;
    while(method!='!'){
        int num;
        cin>>method;
        if(method == '+'){
            cin>>num;
            deq.push_front(num);
        }
        if(method == '-'){
            cin>>num;
            deq.push_back(num);
        }
        if(method == '*'){
            if(deq.size() == 1){
                cout << deq.back() * 2 << endl;
                deq.pop_back(); 
            }
            else if (deq.empty()){
                cout<<"error"<<endl;
            }
            else{
                int sum = deq.back() + deq.front(); 
                deq.pop_back(); 
                deq.pop_front();
                cout << sum << endl;
            }
        }
    }
}

File: 1 lab/j.py
a = [0] * 100
front, back = 50, 49
while True:
    method = input().split()
    if not method:
        continue
    if method[0] == "!":
        break
    if method[0] == "+":
        front -= 1
        a[front] = int(method[1])
    elif method[0] == "-":
        back += 1
        a[back] = int(method[1])
    elif method[0] == "*":
        if front < back:
            print(a[front] + a[back])
            a[front] = 0
            a[back] = 0
            front += 1
            back -= 1
        elif front == back:
            print(a[front] + a[back])
            a[front] = 0
            a[back] = 0
        else:
            print("error")
